src/utils/graphql/queries/getApplications.query.ts

```diff
@@ -1,8 +1,10 @@
 import { gql } from '@apollo/client'
 
 export default gql`
-  query getApplications {
-    applications {
+  query getApplications($code: String!) {
+    applications(
+      filter: { template: { code: { equalTo: $code }, and: { status: { equalTo: AVAILABLE } } } }
```

Comment:
I don't think that this is dependent on if the filter is passed or not. We would need to query GraphQL To get all possible `labels` for certain columns. Example: Action, Stage, Status. This would be available for filter selection at some point, and also for checking for valid filter values (I guess)

Reply:
What I mean is, say for a query like:
```
query MyQuery($status: ApplicationStatus) {
  applications(filter: { status: { equalTo: $status } }) {
    nodes {
      name
      isActive
      status
      stage
    }
  }
}

```

If we don't supply a value for `$status`, what happens? What we _want_ to happen is that it would ignore that filter and return ALL applications, but it seems like GraphQL throws an error. So we'll need to figure out how to solve this problem. I guess the whole "filter" string would need be dynamically created and inserted into the GraphQL string, right?

Reply:
Oh, I see. I wasn't getting this idea. You can pass optional parameters to the GraphQL query, so in case the parameter isn't defined you won't add to the query. I'm not sure how we would do that with different filters. I'll see if I can come up with a solution for that - anyway it's not included in this issue.

Reply:
I think I've figured out how this can work.

We need to enable `connectionFilterAllowNullInput` in the advanced filtering plugin (see [here](https://github.com/graphile-contrib/postgraphile-plugin-connection-filter#connectionfilterallownullinput))

Then we can just set the value of the variable to `null` and it will ignore that filter. Nice ;)

Reply:
Cool! As agreed this would be included as part of issue #226.